app/models/service_order.rb

```diff
@@ -5,14 +5,15 @@ class ServiceOrder < ActiveRecord::Base
 
   belongs_to :tenant
   belongs_to :user
-  has_many :miq_requests, :dependent => :nullify
+  has_many :miq_requests
```

Comment:
The `before_destroy` has to be registered before the `:dependent => :nullify` for it to work properly.